07week/ticTacToe/script.js

```diff
@@ -1,27 +1,105 @@
 'use strict';
 
 class TicTacToe extends React.Component {
-  constructor(props) {
-    super(props);
-  }
+        constructor(props) {
+            super(props);
+            this.state = {
+                turn: 'X',
+                grid: [
+                  [null, null, null],
+                  [null, null, null],
+                  [null, null, null]
+              ]
+            }
+            this.takePlayerTurn = this.takePlayerTurn.bind(this);
+        }
+
+        takePlayerTurn(event) {
+            // console.log("hello world");
+            // console.log(this.state.turn);
+            // console.log(this.state.grid);
+
+            let cellInput = event.target.getAttribute("data-cell");
+            let playerTurn = this.state.turn;
+            let gameBoard = this.state.grid;
+            let newState = this.state.turn;
+
+            //console.log(cellInput);
+
+            // can add playerTurn to the array element (onClick)
+            if (cellInput === "0") {
```

Comment:
if you store the board in the state as a 1 dimensional array it would have been less verbose. but this also works.